editor/src/components/editor/hook-utils.ts

```diff
@@ -26,3 +26,22 @@ export function useForceUpdate() {
   const [, forceUpdate] = React.useReducer((c) => c + 1, 0)
   return forceUpdate
 }
+
+export function useDelayedValueHook(inputValue: boolean, delayMs: number): boolean {
+  const [returnValue, setReturnValue] = React.useState(inputValue)
+  React.useEffect(() => {
+    let timerID: any = undefined
+    if (inputValue) {
+      // we do not delay the toggling if the input value is true
```

Comment:
This custom logic is specific to our use of the value in the `editor-component`, and so shouldn't live here. For the sake of this PR, I'd say we should move this code from here into `editor-component`, otherwise this code is presenting itself as being far more generic than it actually is.

Reply:
@Rheeseyb next up is using this for the right pane (eg inspector) and then things that should similarly animate (eg sections in the left menu). What about the behaviour is specific, and how can we make it more generic?

Reply:
@Rheeseyb Hey Rheese any suggestions to make it more generic as @maltenuhn mentioned?

Reply:
I think we shouldn't try to make it more generic than it is. when you work on the right pane animation, you should start with a copy-pasted version, and tailor it for the inspector's needs. once that is done, we can look at the two hooks, and see what are the similarities, and how (and if) it can be merged into one hook.

Reply:
Yeah, as Balazs says we don't need to make this more generic right now. I'm just suggesting that we please move the code into the source file where it is used so that it isn't being presented as being directed.

Reply:
Done!